File: crave-on-api/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'category_id',
        'name',
        'slug',
        'description',
        'price',
        'image_url',
        'is_available',
        'stock',
    ];

    protected function casts(): array
    {
        return [
            'price'        => 'decimal:2',
            'is_available' => 'boolean',
            'stock'        => 'integer',
        ];
    }

    // ── Relationships ─────────────────────────────────────

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function orderItems(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    // ── Query Scopes ──────────────────────────────────────

    // Product::available()->get()
    public function scopeAvailable($query)
    {
        return $query->where('is_available', true);
    }

    // Product::byCategory('iced')->get()
    public function scopeByCategory($query, string $slug)
    {
        return $query->whereHas('category', fn($q) =>
            $q->where('slug', $slug)
        );
    }
}

File: crave_on_mobile/lib/models/product.dart
import 'category.dart';

class Product {
  final int       id;
  final String    name;
  final String    slug;
  final String?   description;
  final double    price;
  final String?   imageUrl;
  final bool      isAvailable;
  final int       stock;
  final Category? category;

  Product({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    required this.price,
    this.imageUrl,
    required this.isAvailable,
    required this.stock,
    this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id:          json['id'],
      name:        json['name'],
      slug:        json['slug'],
      description: json['description'],
      price:       (json['price'] as num).toDouble(),
      imageUrl:    json['image_url'],
      isAvailable: json['is_available'] ?? true,
      stock:       json['stock'] ?? 0,
      category:    json['category'] != null
          ? Category.fromJson(json['category'])
          : null,
    );
  }
}